nvme.c

```diff
@@ -275,65 +280,78 @@ static int get_telemetry_log(int argc, char **argv, struct command *cmd, struct
 	}
 
 	cfg.host_gen = !!cfg.host_gen;
+	hdr = malloc(bs);
+	page_log = malloc(bs);
+	if (!hdr || !page_log) {
+		fprintf(stderr, "Failed to allocate %zu bytes for log\n", bs);
+		err = ENOMEM;
+		goto free_mem;
+	}
 
-	hdr = malloc(4096);
-	memset(hdr, 0, 4096);
-
-	err = nvme_get_telemetry_log(fd, hdr, cfg.host_gen, cfg.ctrl_init, 4096, 0);
+	memset(hdr, 0, bs);
+	err = nvme_get_telemetry_log(fd, hdr, cfg.host_gen, cfg.ctrl_init, bs, 0);
 	if (err) {
 		fprintf(stderr, "NVMe Status:%s(%x)\n",
 			nvme_status_to_string(err), err);
 		fprintf(stderr, "Failed to aquire telemetry header %d!\n", err);
-		free(hdr);
-		goto close_output;
+		close(output);
```

Comment:
Could you please return close(output) into close_output or close_fd? Fallthrough label flow will close it anyway.